### DISCUSSION TITLE
Custom login page with Auth.js - am I doing this right?

### DISCUSSION THREAD

**Discussion by aaronmarsden** [nextauthjs/next-auth#8239] [Help]
Hey all,

I'm using Auth.js for authentication on the SvelteKit app. The auth method is a magic-link email.

In the [docs on custom sign in pages](https://next-auth.js.org/configuration/pages#email-sign-in), they mention that you can use the signIn() function on the custom login / signup page which will "handle obtaining the CRSF token for you."

So basically, I've just imported the signIn function in my +page.svelte for the custom sign in page, created an input that binds to a value, and assigned a function on my button that will do some validation and run the signIn() function after the checks.

signIn('email', { email: userEmail });

Is this correct? Does this leave me open to any security vulnerabilities? I assume it's OK as the signIn() function is used in other examples client side, but they don't have any docs specifically for custom sign in pages as they relate to SvelteKit.

Any help is much appreciated!

**Comment by 4Furki4** [comment]
It seems working well based on my experience using React in Next.js so far. I created my custom login page using signIn function for credential login and oAuth login.
